### ISSUE TITLE
feat: add custom segments configuration

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#22]
## Problem

Currently, segments are hardcoded: `ui`, `model`, `lib`, `api`, `config`, `assets`.

Different architectures may use additional segments like `services`, `stores`, `hooks`, `composables`, `types`, etc.

There's no way to customize the list of valid segments.

## Use Case

Architecture with custom segments:
```
features/
└── booking/
    ├── ui/
    ├── model/
    ├── services/    # Custom segment
    ├── hooks/       # Custom segment
    └── index.ts
```

Current behavior:
- `services/` and `hooks/` are not recognized as valid segments
- Public API validation doesn't work correctly for them

## Proposed Solution

Add global `segments` configuration option:

```js
featureSliced({
  segments: {
    mode: 'extend',  // or 'override'
    list: ['services', 'stores', 'hooks'],
  },
})
```

### Modes

**extend** — adds custom segments to defaults (ui, model, lib, api, config, assets)
**override** — completely replaces the default set

### Strict Validation

Unknown segments in code should produce an error in both modes.

## Expected Behavior

```js
// With config: segments: { mode: 'extend', list: ['services'] }

import { bookingService } from '~/features/booking/services' // ❌ ERROR - deep import
import { bookingService } from '~/features/booking'          // ✅ OK - through Public API

// Unknown segment
import { something } from '~/features/booking/unknown'       // ❌ ERROR - unknown segment
```

## Affected Rules

- `public-api` — segment validation and extraction

**Comment by conarti** [comment]
Implemented in 2.0.0 (custom segments via the `segments` option of `featureSliced()`, PR #24). Available now as `npm i -D @conarti/eslint-plugin-feature-sliced@next` (2.0.0-rc.16); it will ship in 2.0.0.
